test/test_foreach.py

```diff
@@ -229,11 +229,7 @@ def test_parity(self, device, dtype, op, noncontiguous, inplace):
                         **sample.kwargs,
                     )
             except Exception as e:
-                with (
-                    self.assertRaisesRegex(type(e), re.escape(str(e).splitlines()[0]))
-                    if not (op.has_no_in_place or not op.supports_out)
-                    else self.assertRaises(type(e))
-                ):
+                with self.assertRaises(type(e)):
```

Comment:
I attempted to preserve some form of the regex text check here, but this test is used across several different underlying CUDA implementations with different error messages, so there was no consistent pattern to rely on.

Reply:
This looks fine to me. If there is not a reliable pattern that can be applied to all different cases then we shouldn't have the extra condition of checking the message contents for uniform wording.

Reply:
Do we know what messages we're no longer checking? To my knowledge, this was intended to ensure that add and _foreach_add or <op> and _foreach_<op> would fail similarly.

Reply:
@janeyx99 Yes, there are several with functions that had non-matching error messages, and some of them were not trivial to fixup. Some quick hits:

* `aten/src/ATen/native/cuda/ForeachUnaryOp.cu`: lines 87-202 define a bunch of generic function templates which all register themselves with the dispatcher under the name `foreach_unary_op_cuda`.  That name appears in the error messages instead of the correct function name.
  * It's possible to pass the desired function name as a template parameter, but the C++ is pretty ugly (`static constexpr char[]` has to be used to store the function name, before passing it through the template).
  * Similar patterns appear to be in most (all?) of the `Foreach*Op.cu` files.
* More trivial: `UnaryOpsKernel.cu` et. al. have functions registered as `exp_cuda`, `expm1_cuda`, etc. These error messages could be fixed by dropping the `_cuda` suffix.
* An additional set of non-matching error messages occurs in `aten/src/ATen/native/UnaryOps.cpp`, lines 259-291.

I considered attempting to make all the error messages match a common pattern, but it seemed like a very extensive set of changes (I'm not convinced this list captures the full set of changes).  It seemed worthwhile, at least in the short term, to validate that an error was correctly thrown in all cases, even if the text didn't match. Doing so helped me catch the bugs fixed in the other two MRs in the stack.